tests/xfstests/run.pm

```diff
@@ -55,90 +44,9 @@ my $STATUS_LOG   = '/opt/status.log';
 my $INST_DIR     = '/opt/xfstests';
 my $LOG_DIR      = '/opt/log';
 my $KDUMP_DIR    = '/opt/kdump';
-my $MAX_TIME     = get_var('XFSTESTS_SUBTEST_MAXTIME') || 2400;
 my $FSTYPE       = get_required_var('XFSTESTS');
-
-# Create heartbeat script, directories(Call it only once)
-sub test_prepare {
-    my $redir  = " >> /dev/$serialdev";
-    my $script = <<END_CMD;
-#!/bin/sh
-rm -f $HB_DONE_FILE $HB_EXIT_FILE
-declare -i c=0
-while [[ ! -f $HB_EXIT_FILE ]]; do
-    if [[ -f $HB_DONE_FILE ]]; then
-        c=0
-        echo "$HB_DONE" $redir
-        sleep 2
-    elif [[ \$c -ge $HB_INTVL ]]; then
-        c=0
-        echo "$HB_PATN" $redir
-    else
-        c+=1
-    fi
-    sleep 1
-done
-END_CMD
-    assert_script_run("cat > $HB_SCRIPT <<'END'\n$script\nEND\n( exit \$?)");
-    assert_script_run("mkdir -p $KDUMP_DIR $LOG_DIR");
-}
-
-# Start heartbeat, setup environment variables(Call it everytime SUT reboots)
-sub heartbeat_start {
-    enter_cmd(". ~/.xfstests; nohup sh $HB_SCRIPT &");
-}
-
-# Stop heartbeat
-sub heartbeat_stop {
-    send_key 'ret';
-    assert_script_run("touch $HB_EXIT_FILE");
-}
-
-# Wait for heartbeat
-sub heartbeat_wait {
-    # When under heavy load, the SUT might be unable to send
-    # heartbeat messages to serial console. That's why HB_TIMEOUT
-    # is set to 200 by default: waiting for such tests to finish.
-    my $ret = wait_serial([$HB_PATN, $HB_DONE], $HB_TIMEOUT);
-    if ($ret) {
-        if ($ret =~ /$HB_PATN/) {
-            return ($HB_PATN, '');
-        }
-        else {
-            my $status;
-            send_key 'ret';
-            my $ret = script_output("cat $HB_DONE_FILE; rm -f $HB_DONE_FILE");
-            $ret =~ s/^\s+|\s+$//g;
-            if ($ret == 0) {
-                $status = 'PASSED';
-            }
-            elsif ($ret == 22) {
-                $status = 'SKIPPED';
-            }
-            else {
-                $status = 'FAILED';
-            }
-            return ($HB_DONE, $status);
-        }
-    }
-    return ('', 'FAILED');
-}
```

Comment:
This will introduce two new problems.
1. No "SKIPPED" status
such as: http://10.67.133.102/tests/236#step/1_test_result/40 
btrfs/142 marked as "PASSED", but its output is "Not run: btrfs/142" in test log

2. Cannot catch some "FAILED" status
such as http://10.67.133.102/tests/234#step/1_test_result/11
The result of btrfs/11 is "PASSED", but actually its "FAILED" in test log

Reply:
Nice finding, I double check in verify run see similar issue.
Here test fail but checking result all pass: https://openqa.suse.de/tests/5894403#step/run/427

Reply:
Nonsense! No it is not introducing already broken test, which returning 0 when it failed. So please report this issues which is failing since ever ... 
btrfs/11: https://openqa.suse.de/tests/5990847#step/run/236
btrfs/142 : https://openqa.suse.de/tests/5990849#step/run/826
Again is really disturbing that you point out test which is failing with old code!

Reply:
### With old code:
btrfs/11 is **FAILED**: https://openqa.suse.de/tests/5990847#step/run/236
And test report is **FAILED**: https://openqa.suse.de/tests/5990847#step/1_test_result/11

btrfs/142 is **NOT RUN**: https://openqa.suse.de/tests/5990849#step/run/826
And test report is **NOT RUN**: https://openqa.suse.de/tests/5990849#step/1_test_result/40


### In your VR:
btrfs/11 is **FAILED**: http://10.67.133.102/tests/234#step/run/172
But test report is **PASSED**: http://10.67.133.102/tests/234#step/1_test_result/11

btrfs/142 is **NOT RUN**: http://10.67.133.102/tests/236#step/run/666
But test report is **PASSED**: http://10.67.133.102/tests/236#step/1_test_result/40

Reply:
The test could run with whatever code, I don't understand why you don't use CASEDIR.
I will test it again and try to reproduce your issue, but this are my VR on osd:
btrfs/11 https://openqa.suse.de/tests/5894387#step/run/260
btrfs/142 https://openqa.suse.de/tests/5894389#step/run/956

Reply:
You have reproduced it. They only show "PASS" in test result, but "NOT RUN" according to test log.

Reply:
Tests which output [not run] should mark as "SKIPPED" and gray color(not green) in test result

Reply:
Test is marked base on what result it returns `Passed all 1 tests`

Reply:
In your code, test is marked based on `$@ which return by eval` but not `Passed all 1 tests`

Reply:
True, there is just pass/fail. I will change that.